frontend/app/components/work-packages/wp-attachments-formattable-field/wp-attachments-formattable.directive.ts

```diff
@@ -0,0 +1,131 @@
+
+    }
+
+    public handleDrop = (evt: DragEvent): void => {
+        evt.preventDefault();
+        evt.stopPropagation();
+
+        const textarea: ng.IAugmentedJQuery = this.$element.find('textarea');
+        this.viewMode  = (textarea.length > 0) ? ViewMode.EDIT : ViewMode.SHOW;
+
+        const workPackage: WorkPackageResource = this.$scope.workPackage;
+        const dropData: DropModel = new DropModel(this.$location, evt.dataTransfer, workPackage);
+      
+        var description: any;
+
+        if (this.viewMode === ViewMode.EDIT) {
+            description = new EditorModel(textarea, new MarkupModel());
+        }
+        else {
+            description = new FieldModel(workPackage, new MarkupModel());
+        }
+
+        if (angular.isUndefined(dropData.webLinkUrl) && angular.isUndefined(dropData.files))
+            return;
+
+        if(dropData.isUpload){
+            if(dropData.filesAreValidForUploading()) {
+                if(!dropData.isDelayedUpload) {
+                    this.wpAttachments.upload(workPackage, dropData.files).then(() => {
+                            this.wpAttachments.load(workPackage,true).then((updatedAttachments: any) => {
+                                if (angular.isUndefined(updatedAttachments))
+                                    return;
+
+                                updatedAttachments.sort(function(a, b){
+                                    return a.id > b.id ? 1 : -1;
+                                });
+
+                                if (dropData.filesCount === 1) {
+                                    const currentFile: SingleAttachmentModel =
+                                      new SingleAttachmentModel(updatedAttachments[updatedAttachments.length - 1]);
+                                    description.insertAttachmentLink(
+                                      currentFile.url,
+                                      (currentFile.isAnImage) ? InsertMode.INLINE : InsertMode.ATTACHMENT);
+                                }
+                                else if (dropData.filesCount > 1) {
+                                    for (let i: number = updatedAttachments.length - 1;
+                                      i >= updatedAttachments.length - dropData.filesCount;
+                                      i--) {
+                                        description.insertAttachmentLink(
+                                          updatedAttachments[i]._links.downloadLocation.href,
+                                          InsertMode.ATTACHMENT,
+                                          true);
+                                  }
+                                }
+                                description.save();
+                            },(err)=>{
+                              console.log("error while reloading attachments",err)
+                            });
+                    }, function(err) {
+                        console.log(err);
```

Comment:
Please use the `NotificationsService` to provide error handling for cases where attachments cannot be uploaded. Never only rely on the console for errors unless the error is of no use for users.